vertical-pod-autoscaler/pkg/recommender/logic/recommender.go

```diff
@@ -147,3 +149,31 @@ func CreatePodResourceRecommender() PodResourceRecommender {
 		lowerBoundEstimator,
 		upperBoundEstimator}
 }
+
+// MapToListOfRecommendedContainerResources converts the map of RecommendedContainerResources into a stable sorted list
+// This can be used to get a stable sequence while ranging on the data
+func MapToListOfRecommendedContainerResources(resources RecommendedPodResources) *vpa_types.RecommendedPodResources {
```

Comment:
This code is just an extraction as a function of the existing code here:

https://github.com/kubernetes/autoscaler/blob/eded7ecf2b84438a30d826d3ba4b93d00454ada8/vertical-pod-autoscaler/pkg/recommender/routines/recommender.go#L147-L168